web/modules/custom/dbcdk_community/src/Profile/ProfileRepository.php

```diff
@@ -0,0 +1,357 @@
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  protected function findProfile(array $filter) {
+    $profile = $this->profileApi->profileFindOne($this->processFilter($filter));
+    if (!empty($profile)) {
+      $profile = $this->enrichProfile($profile);
+    }
+    return $profile;
+  }
+
+  /**
+   * Get all profiles matching a filter.
+   *
+   * @param array $filter
+   *   The filter to use.
+   *
+   * @return \DBCDK\CommunityServices\Model\Profile[]
+   *   The matching profiles.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  public function getProfiles(array $filter = []) {
+    return (array) $this->profileApi->profileFind($this->processFilter($filter));
+  }
+
+  /**
+   * Count the number of profiles matching a filter.
+   *
+   * @param array $filter
+   *   The filter to use.
+   *
+   * @return int|NULL
+   *   The number of profiles matching the filter. NULL if no count is returned.
+   */
+  public function countProfiles(array $filter = []) {
+    $result = $this->profileApi->profileCount($this->processFilter($filter));
+    return (isset($result['count'])) ? $result['count'] : NULL;
+  }
+
+  /**
+   * Enrich a "flat" profile object.
+   *
+   * This adds details about community roles and quarantines.
+   *
+   * @param ModelProfile $profile
+   *   The original profile object.
+   *
+   * @return Profile
+   *   The profile object enriched with details ab
+   */
+  public function enrichProfile(ModelProfile $profile) {
+    $profile = new Profile($profile);
+    $profile->setCommunityRoles((array) $this->profileApi->profilePrototypeGetCommunityRoles($profile->getId()));
+    $profile->setQuarantines((array) $this->profileApi->profilePrototypeGetQuarantines($profile->getId()));
+    return $profile;
+  }
+
+  /**
+   * Get profiles with an active quarantine at a specific time.
+   *
+   * @param DateTime $time
+   *   The time to check quarantine date ranges against.
+   * @param array $profiles_filter
+   *   An array containing filter arguments.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   *
+   * @return \DBCDK\CommunityServices\Model\Profile[] $profiles
+   *   An array of Profile objects with an active quarantine.
+   */
+  public function getQuarantinedProfiles(DateTime $time, array $profiles_filter = []) {
+    $quarantines = $this->getActiveQuarantines($time);
+
+    // Use the array of quarantined ids as "where" arguments.
+    $profiles_filter['where']['or'] = $this->getQuarantinedProfilesFilter($quarantines);
+
+    return (array) $this->profileApi->profileFind($this->processFilter($profiles_filter));
+  }
+
+  /**
+   * Count the number of quarantined profiles matching a filter.
+   *
```

Comment:
Should we include a description saying that it's not a simple "count" request but that it actually fetches all profiles and then counts them?